File: frontend/src/types/paginacao.ts
export interface ListaPaginada<T> {
  dados: T[];
  total: number;
  pagina: number;
  limite: number;
}

export interface FiltroPaginacao {
  pagina?: number;
  limite?: number;
  busca?: string;
  status?: string;
  tipo?: string;
  finalidade?: string;
}

File: backend/internal/core/domain/paginacao.go
package domain

type FiltroPaginacao struct {
	Pagina     int
	Limite     int
	Busca      string
	Status     string
	Tipo       string
	Finalidade string
	UsuarioID  *string
}

type ListaPaginada[T any] struct {
	Dados  []T `json:"dados"`
	Total  int `json:"total"`
	Pagina int `json:"pagina"`
	Limite int `json:"limite"`
}
